File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.php
class Solution {

  /**
    * @param Integer[] $nums
    * @return Boolean
    */
  function check($nums) {
    $nums []= $nums[0];
    for ($i = 1, $fold = false; $i < count($nums); $i++) {
      if ($nums[$i] < $nums[$i - 1]) {
        if ($fold) return false;
        $fold = true;
      }
    }
    return true;
  }
}

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.py
class Solution:
  def check(self, nums: List[int]) -> bool:
    nums.append(nums[0])
    fold = False
    for i in range(1, len(nums)):
      if nums[i] < nums[i - 1]:
        if fold:
          return False
        fold = True
    return True